src/views/pages/api-test/TestPost.js

```diff
@@ -0,0 +1,136 @@
+import { useState } from "react";
+import {
+    Box,
+    Button,
+    Card,
+    CardActions,
+    CardContent,
+    CardHeader,
+    Chip,
+    Divider,
+    MenuItem,
+    Paper,
+    Select,
+    Stack,
+    TextField,
+    Typography
+} from "@mui/material";
+
+import useApi from "hooks/useApi";
+import testApi from "api/test";
+import { styled } from "@mui/material/styles";
+import StateHandler from "ui-component/StateHandler";
+
+const Item = styled(Paper)(({ theme }) => ({
+    backgroundColor: theme.palette.mode === "dark" ? "#1A2027" : "#fff",
+    ...theme.typography.body2,
+    padding: theme.spacing(1),
+    textAlign: "center",
+    color: theme.palette.text.secondary
+}));
+
+const TestPost = () => {
+    // This API performs an operation between two numbers
+    // Parameters: n1, n2, op
+    // Supported ops: add, sub, mul, div
+    const calcApi = useApi(testApi.postTest);
+    const [calcState, setCalcState] = useState({
+        n1: "",
+        n2: "",
+        op: "add"
+    });
+
+    const handleTextInputChange = (event) => {
+        console.log("handleTextInputChange:", event.target);
+        const value = event.target.value;
+        setCalcState({
+            ...calcState,
+            [event.target.name]: value
+        });
+        console.log("calcState:", calcState);
```

Comment:
when onChange triggers, I update the state based on the element's name and new value.
If only the input element "n1" was changed, then only the "n1" key in the state dictionary gets updated.

IMPORTANT: Notice how when setCalcState is called, all elements that use calcState get updated automatically with the new values.